lib/ReactViews/DragDropFile.tsx

```diff
@@ -12,36 +12,27 @@ import Result from "../Core/Result";
 import CatalogMemberMixin, { getName } from "../ModelMixins/CatalogMemberMixin";
 import MappableMixin from "../ModelMixins/MappableMixin";
 import addUserFiles from "../Models/Catalog/addUserFiles";
-import { BaseModel } from "../Models/Definition/Model";
-import Styles from "./drag-drop-file.scss";
-import { WithViewState, withViewState } from "./Context";
+import type { BaseModel } from "../Models/Definition/Model";
 import { raiseFileDragDropEvent } from "../ViewModels/FileDragDropListener";
+import { useViewState } from "./Context";
+import Styles from "./drag-drop-file.scss";
 
-interface PropsType extends WithTranslation, WithViewState {}
-
-@observer
-class DragDropFile extends Component<PropsType> {
-  target: EventTarget | undefined;
-
-  constructor(props: PropsType) {
-    super(props);
-    makeObservable(this);
-  }
+const DragDropFile: FC = observer(() => {
+  const viewState = useViewState();
+  const targetRef = useRef<EventTarget>();
```

Comment:
I think I've seen something that some future React version requires an argument to `useRef()`, is it worth adding the `(null)` argument now since the code is already being changed here?

Reply:
yeah it is in react 19, will handle that when we get to upgrading to react 19